lib/Auth.js

```diff
@@ -68,21 +68,45 @@ function useAuthProvider() {
 
         await fetchUserProfileAndStartClient();
 
-        // Loop through all existing authentication providers and sign in
-        Object.entries(activeAuthentications).forEach(([type, authentication]) => {
-            // Skip the default Matrix auth provider
-            if (type === 'matrix') return;
-            authentication.signin(username, password).catch(() => { });
-        });
+        await getAccessTokensFromAccountData(username, password);
 
         return matrixAuthenticationResponse;
     };
 
+    const getAccessTokensFromAccountData = async (username, password) => {
+        // we will use `getAccountDataFromServer` instead of `getAccountData` as we can not tell for sure if the sync is completely done and dont want to wait for that. So potentially this call is avoidable but it is more manageable to handle it like this.
+        const accessTokens = await getAuthenticationProvider('matrix').getMatrixClient().getAccountDataFromServer('dev.medienhaus.spaces.accesstokens') || {};
+        // fetch did not worked or this accountData state event is not set yet. In either way we want to create it.
+        let dataChanged = false;
+
+        for await (const [type, authentication] of Object.entries(activeAuthentications)) {
+            // Skip the default Matrix auth provider
+            if (type === 'matrix') continue;
+
+            if (accessTokens[type]) {
+                authentication.addToken(accessTokens[type]);
+                const tokenValidation = await authentication.validateToken();
+                if (tokenValidation) {
+                    // everything seems to work out as expected we will cut here
```

Comment:
Continuing the discussion from the main thread:

Something is wrong here: `if (tokenValidation) {` ... so _if_ the token is valid, then yes, everything seems to work and we can continue.

But there's no `} else {` in line 92. This code is not doing anything _if_ the token from the Matrix account data is expired or wrong.